Abstract enums are not initialized

Hello,

when trying to populate a class that uses an abstract enum, an error is produced. Fixing the issue is possible in the FieldPopulator.generateRandomValue method by adding
```
else if (isEnumType(fieldType))
{
     value = this.beanPopulator.doPopulateBean(fieldType, context);
 }
```
Randomizer:
```
    EnhancedRandom random = EnhancedRandomBuilder.aNewEnhancedRandomBuilder()
                .scanClasspathForConcreteTypes(true)
                .build();

        TestObject testObject = random.nextObject(TestObject.class);
        System.out.println(testObject);
```

Enum:
```
public enum TestEnum
{
    VAL1("Val1")
    {
        @Override
        public String test()
        {
            return "1";
        }
    },
    VAL2("Val2")
    {
        @Override
        public String test()
        {
            return "2";
        }
    };

    private final String displayString;

    TestEnum(String displayString)
    {
        this.displayString = displayString;
    }

    public abstract String test();
}
```
Class:
```
public class TestObject
{
    public TestEnum testEnum;
}
```
Error:
```
io.github.benas.randombeans.api.ObjectGenerationException: Unable to generate a random instance of type class TestObject

	at io.github.benas.randombeans.EnhancedRandomImpl.doPopulateBean(EnhancedRandomImpl.java:122)
	at io.github.benas.randombeans.EnhancedRandomImpl.nextObject(EnhancedRandomImpl.java:72)
	at GenericCloneTestSingle.testTestEnum(GenericCloneTestSingle.java:79)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.intellij.junit4.JUnit4TestRunnerUtil$IgnoreIgnoredTestJUnit4ClassRunner.runChild(JUnit4TestRunnerUtil.java:365)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:117)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:253)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:84)
Caused by: io.github.benas.randombeans.api.ObjectGenerationException: Unable to find a matching concrete subtype of type: class TestEnum
	at io.github.benas.randombeans.FieldPopulator.generateRandomValue(FieldPopulator.java:100)
	at io.github.benas.randombeans.FieldPopulator.populateField(FieldPopulator.java:79)
	at io.github.benas.randombeans.EnhancedRandomImpl.populateField(EnhancedRandomImpl.java:153)
	at io.github.benas.randombeans.EnhancedRandomImpl.populateFields(EnhancedRandomImpl.java:144)
	at io.github.benas.randombeans.EnhancedRandomImpl.doPopulateBean(EnhancedRandomImpl.java:118)
	... 24 more
```

--- a/random-beans/src/main/java/io/github/benas/randombeans/FieldPopulator.java
+++ b/random-beans/src/main/java/io/github/benas/randombeans/FieldPopulator.java
@@ -94,7 +94,7 @@ class FieldPopulator {
         } else if (isMapType(fieldType)) {
             value = mapPopulator.getRandomMap(field, context);
         } else {
-            if (scanClasspathForConcreteTypes && isAbstract(fieldType)) {
+            if (scanClasspathForConcreteTypes && isAbstract(fieldType) && !isEnumType(fieldType) /*enums can be abstract, but can not inherit*/) {
                 Class<?> randomConcreteSubType = randomElementOf(filterSameParameterizedTypes(getPublicConcreteSubTypesOf(fieldType), fieldGenericType));
                 if (randomConcreteSubType == null) {
                     throw new ObjectGenerationException("Unable to find a matching concrete subtype of type: " + fieldType);

--- a/random-beans/src/test/java/io/github/benas/randombeans/EnhancedRandomImplTest.java
+++ b/random-beans/src/test/java/io/github/benas/randombeans/EnhancedRandomImplTest.java
@@ -239,6 +239,15 @@ public class EnhancedRandomImplTest {
         assertThat(foo.getBar().getName()).isNotEmpty();
     }
 
+    @Test
+    public void whenScanClasspathForConcreteTypesIsEnabled_thenShouldPopulateAbstractEnumeration() {
+        EnhancedRandom random = EnhancedRandomBuilder.aNewEnhancedRandomBuilder().scanClasspathForConcreteTypes(true).build();
+
+        ClassUsingAbstractEnum randomValue = random.nextObject(ClassUsingAbstractEnum.class);
+
+        then(randomValue.getTestEnum()).isNotNull();
+    }
+
     @Test(expected = IllegalArgumentException.class)
     public void whenSpecifiedNumberOfBeansToGenerateIsNegative_thenShouldThrowAnIllegalArgumentException() {
         enhancedRandom.objects(Person.class, -2);

--- a/random-beans/src/test/java/io/github/benas/randombeans/beans/AbstractEnum.java
+++ b/random-beans/src/test/java/io/github/benas/randombeans/beans/AbstractEnum.java
@@ -1,0 +1,43 @@
+/**
+ * The MIT License
+ *
+ *   Copyright (c) 2016, Mahmoud Ben Hassine (mahmoud.benhassine@icloud.com)
+ *
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ */
+package io.github.benas.randombeans.beans;
+
+public enum AbstractEnum {
+    VAL1() {
+
+        @Override
+        public String test() {
+            return "1";
+        }
+    },
+    VAL2() {
+
+        @Override
+        public String test() {
+            return "2";
+        }
+    };
+
+    public abstract String test();
+}

--- a/random-beans/src/test/java/io/github/benas/randombeans/beans/ClassUsingAbstractEnum.java
+++ b/random-beans/src/test/java/io/github/benas/randombeans/beans/ClassUsingAbstractEnum.java
@@ -1,0 +1,32 @@
+/**
+ * The MIT License
+ *
+ *   Copyright (c) 2016, Mahmoud Ben Hassine (mahmoud.benhassine@icloud.com)
+ *
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ */
+package io.github.benas.randombeans.beans;
+
+import lombok.Data;
+
+@Data
+public class ClassUsingAbstractEnum {
+
+    private AbstractEnum testEnum;
+}

